### ISSUE TITLE
Problem with owned types for views

### ISSUE THREAD

**Issue by JonasBRasmussen** [efcore/EFCore.NamingConventions#129]
I'm trying to use the library with views, but have run into problems with owned types.

I am testing with EF Core 6.0.2 (making it different from #115) and EFCore.NamingConventions 6.0.0. The following classes shows a minimal example of the problem:

```cs
public class MyView
{
    public long Id { get; set; }

    public Address Address { get; set; }
}

public class Address
{
    public string City { get; set; }
}

public class MyViewConfiguration : IEntityTypeConfiguration<MyView>
{
    public void Configure(EntityTypeBuilder<MyView> builder)
    {
        builder.ToView(nameof(MyView));

        builder.OwnsOne(x => x.Address);
    }
}

public class MyDbContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseNpgsql().UseSnakeCaseNamingConvention();

        base.OnConfiguring(optionsBuilder);
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.ApplyConfiguration(new MyViewConfiguration());

        base.OnModelCreating(modelBuilder);
    }
}
```

This results in the follow exception:
```
Value cannot be null. (Parameter 'name')
   at Microsoft.EntityFrameworkCore.Utilities.Check.NotNull[T](T value, String parameterName)
   at Microsoft.EntityFrameworkCore.Metadata.StoreObjectIdentifier.Table(String name, String schema)
   at Microsoft.EntityFrameworkCore.RelationalKeyExtensions.GetName(IReadOnlyKey key)
   at EFCore.NamingConventions.Internal.NameRewritingConvention.ProcessKeyAdded(IConventionKeyBuilder keyBuilder, IConventionContext`1 context) in C:\Git\GitHub\efcore\EFCore.NamingConventions\EFCore.NamingConventions\Internal\NameRewritingConvention.cs:line 203
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ImmediateConventionScope.OnKeyAdded(IConventionKeyBuilder keyBuilder)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.OnKeyAddedNode.Run(ConventionDispatcher dispatcher)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.DelayedConventionScope.Run(ConventionDispatcher dispatcher)
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ConventionBatch.Run()
   at Microsoft.EntityFrameworkCore.Metadata.Conventions.Internal.ConventionDispatcher.ConventionBatch.Run(IConventionForeignKey foreignKey)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder`1.OwnsOneBuilder[TRelatedEntity](TypeIdentity ownedType, MemberIdentity navigation)
   at Microsoft.EntityFrameworkCore.Metadata.Builders.EntityTypeBuilder`1.OwnsOne[TRelatedEntity](Expression`1 navigationExpression)
   at Demo.MyViewConfiguration.Configure(EntityTypeBuilder`1 builder)    at Microsoft.EntityFrameworkCore.ModelBuilder.ApplyConfiguration[TEntity](IEntityTypeConfiguration`1 configuration)
```

In `ProcessKeyAdded` in `NameRewritingConventions`, the `keyBuilder.Metadata` has debug view `MyViewId PK` and the following `DeclaringEntityType`:
```
EntityType: Address Owned
  Properties: 
    MyViewId (no field, long) Shadow Required PK FK AfterSave:Throw
      Annotations: 
        Relational:ColumnName: my_view_id
    City (string)
      Annotations: 
        Relational:ColumnName: city
  Keys: 
    MyViewId PK
  Foreign keys: 
    Address {'MyViewId'} -> MyView {'Id'} Unique Ownership ToDependent: Address Cascade
      Annotations: 
        Relational:Name: fk_address__my_view_id
  Annotations: 
    RelationshipDiscoveryConvention:NavigationCandidates: System.Collections.Immutable.ImmutableSortedDictionary`2[System.Reflection.PropertyInfo,System.ValueTuple`2[System.Type,System.Nullable`1[System.Boolean]]]
```

There are no errors if I do not use the library or if I use `ToTable` instead of `ToView`.

**Comment by roji** [comment]
Thanks for filing this and for the repro. I'm concentrating on other things at the moment but will be back to look at this.

**Comment by JonasBRasmussen** [comment]
@roji How's your time looking? 😉

**Comment by roji** [comment]
@JonasBRasmussen apologies for leaving this issue unanswered for so long. I've come back to it and tested with the latest bits (about to be released as 8.0.0), and I can't reproduce the issue; it's very possible that it has been fixed since you filed it. Below is the program I'm running and the resulting SQL, which looks fine; it would be great if you could try 8.0.0 and report that everything works for you - if not I'll be happy to reopen.

<details>
<summary>Attempted repro</summary>

```c#
await using var ctx = new BlogContext();
await ctx.Database.EnsureDeletedAsync();
await ctx.Database.EnsureCreatedAsync();

Console.WriteLine(ctx.Set<MyView>().Where(v => v.Address.City == "Berlin").ToQueryString());

public class BlogContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseNpgsql("Host=localhost;Username=test;Password=test")
            .UseSnakeCaseNamingConvention()
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<MyView>(builder =>
        {
            // builder.ToView(nameof(MyView));
            builder.ToView(null);
            builder.OwnsOne(x => x.Address);
        });
    }
}

public class MyView
{
    public long Id { get; set; }

    public Address Address { get; set; }
}

public class Address
{
    public string City { get; set; }
}
```

</details>

Query SQL:

```sql
SELECT m.id, m.address_city
FROM my_view AS m
WHERE m.address_city = 'Berlin'
```
